java/src/jmri/jmrit/operations/routes/RouteEditTableModel.java

```diff
@@ -474,9 +474,8 @@ private void setMaxTrainLength(Object value, RouteLocation rl) {
             // warn that train length might be too short
             if (JmriJOptionPane.showConfirmDialog(null, Bundle.getMessage("LimitTrainLength",
                     length, Setup.getLengthUnit().toLowerCase(), rl.getName()),
-                    Bundle
-                            .getMessage("WarningTooShort"),
-                    JmriJOptionPane.OK_CANCEL_OPTION) != JmriJOptionPane.CANCEL_OPTION) {
+                    Bundle.getMessage("WarningTooShort"),
+                    JmriJOptionPane.OK_CANCEL_OPTION) == JmriJOptionPane.CANCEL_OPTION) {
```

Comment:
I'm not sure, but if I remember correctly, the correct fix is
`JmriJOptionPane.OK_CANCEL_OPTION) != JmriJOptionPane.OK_OPTION)`

I think JmriJOptionPane returns another value if the window is closed instead of pressing cancel.

Reply:
Thanks, I agree with you.  I was too quick to return the code to its previous state, which wasn't quite correct.